base_report_to_printer/report.py

```diff
@@ -53,6 +71,9 @@ def get_pdf(self, cr, uid, ids, report_name, html=None,
         report = self._get_report_from_name(cr, uid, report_name)
         behaviour = report.behaviour()[report.id]
         printer = behaviour['printer']
-        if behaviour['action'] == 'server' and printer and document:
+        can_print_report = self._can_print_report(cr, uid, ids,
+                                                  behaviour, printer, document,
+                                                  context=context)
+        if can_print_report:
```

Comment:
@guewen I wonder if we should not restore the context modification with `must_skip_send_to_printer` in case someone does something mutliple/bad overriding of `get_pdf`. It will add some defensiveness to the code.